Automattic/jetpack
# VideoPress: Provide the list of allowed video extensions on the client initial state

Fixes #26468.

#### Changes proposed in this Pull Request:

* Use the list returned by `get_allowed_mime_types()` to generate a list of `video/*` file extensions
* Provide this list as a parameter on the client initial state data
* Each extension is associated with it's respective mime type, as per @retrofox sugestion [below](https://github.com/Automattic/jetpack/pull/26537#discussion_r984025177)

#### Other information:


#### Jetpack product discussion



N/A.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Run `jetpack docker wp shell`
* On the new shell that opens, run `Automattic\Jetpack\VideoPress\Admin_UI::initial_state()`
* You should be able to see the new parameter, `allowedVideoExtensions`, on the list of parameters returned
* Example:

```
[
    ...
    "allowedVideoExtensions" => [
        "asf" => "video/x-ms-asf",
        "asx" => "video/x-ms-asf",
        "wmv" => "video/x-ms-wmv",
        "wmx" => "video/x-ms-wmx",
        "wm" => "video/x-ms-wm",
        "avi" => "video/avi",
        "divx" => "video/divx",
        "flv" => "video/x-flv",
        "mov" => "video/quicktime",
        "qt" => "video/quicktime",
        "mpeg" => "video/mpeg",
        "mpg" => "video/mpeg",
        "mpe" => "video/mpeg",
        "mp4" => "video/mp4",
        "m4v" => "video/mp4",
        "ogv" => "video/ogg",
        "webm" => "video/webm",
        "mkv" => "video/x-matroska",
        "3gp" => "video/3gpp",
        "3gpp" => "video/3gpp",
        "3g2" => "video/3gpp2",
        "3gp2" => "video/3gpp2",
        "videopress" => "video/videopress",
    ],
]"
```
